pkg/controller/sync.go

```diff
@@ -63,13 +63,21 @@ func (c *Controller) syncContainer(ctx context.Context, log *logrus.Entry,
 	log = log.WithField("container", container.Name)
 	log.Debug("processing container image")
 
+	result, err := c.checker.Container(ctx, log, pod, container, opts)
+	if err != nil {
+		return err
+	}
+
 	err = c.checkContainer(ctx, log, pod, container, containerType, opts)
 	// Don't re-sync, if no version found meeting search criteria
 	if versionerrors.IsNoVersionFound(err) {
 		log.Error(err.Error())
 		return nil
 	}
 	if err != nil {
+		c.metrics.ErrorsReporting(pod.Namespace, pod.Name,
+			container.Name, result.ImageURL,
+		)
```

Comment:
If we move this to L111, then we'll have the image URL available in `result.ImageURL`